base/shell/explorer/startmnucust.cpp

```diff
@@ -66,17 +66,228 @@ VOID OnClearRecentItems()
+    }
+}
+
+typedef void (*STARTMENU_CUSTOM_OPTION)(DWORD, DWORD);
+
+VOID DefaultFunc(DWORD dwItem, DWORD dwState);
+
+static struct
+{
+    PCWSTR OptionName;
+    STARTMENU_CUSTOM_OPTION StartMenu_CustomOption;
+    BOOL IsDirty;
+} ClassicOption[] = {
+    {L"Start_EnableDragDrop", DefaultFunc, FALSE},       // REG_DWORD
+    {L"Start_ShowRun", DefaultFunc, FALSE},              // REG_DWORD
+    {L"StartMenuScrollPrograms", DefaultFunc, FALSE},    // REG_SZ
+    {L"CascadeMyPictures", DefaultFunc, FALSE},          // REG_SZ
+    {L"CascadePrinters", DefaultFunc, FALSE},            // REG_SZ
+    {L"CascadeNetworkConnections", DefaultFunc, FALSE},  // REG_SZ
+    {L"CascadeMyDocuments", DefaultFunc, FALSE},         // REG_SZ
+    {L"CascadeControlPanel", DefaultFunc, FALSE},        // REG_SZ
+    {L"StartMenuFavorites", DefaultFunc, FALSE},         // REG_DWORD
+    {L"StartMenuAdminToolsRoot", DefaultFunc, FALSE},    // REG_SZ
+    {L"ShowSmallIcones", DefaultFunc, FALSE},            // REG_DWORD
+    {L"StartCustomMenus", DefaultFunc, FALSE},           // REG_DWORD
+    {NULL, NULL, FALSE}                                  // Last IsDirty stores the bitmap of all current checkBtn state
+};
+
+VOID DefaultFunc(DWORD dwItem, DWORD dwState)
+{
+    WCHAR buf[128];
+    const WCHAR *ptr = dwState ? L"ON" : L"OFF";
+    wsprintf(buf, L"%s   %s\n", ClassicOption[dwItem].OptionName, ptr);
+
+    MessageBoxW(0, buf, L"Not implemented", MB_OK);
+}
+
+VOID ExecuteCustomOptions(DWORD dwItem, DWORD dwUserOptions, DWORD dwOptSize)
+{
+    for (DWORD idx = 0; idx < dwOptSize; idx++)
+    {
+        if ((dwItem >> idx) & 0x01)
+            ClassicOption[idx].StartMenu_CustomOption(idx, !((dwUserOptions >> idx) & 0x01));
     }
 }
 
+DWORD LoadUserConfData(DWORD *dwLength)
+{
+    HKEY hKey;
+    DWORD iItem;
+    DWORD dwSize;
+    DWORD dwType;
+    DWORD dwStatus = 0;
+    WCHAR chBuffer[16];
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER,
+                     L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Advanced",
+                     0, KEY_QUERY_VALUE,
+                     &hKey) != ERROR_SUCCESS)
+         return 0;
+    for (iItem = 0; ClassicOption[iItem].OptionName != NULL; iItem++)
+    {
+        if (((IDS_DRAG_DROP + iItem) > IDS_SHOW_RUN) &&
+            ((IDS_DRAG_DROP + iItem) < IDS_SHOW_SMALL_ICO) &&
+            ((IDS_DRAG_DROP + iItem) != IDS_SHOW_FAVORITES))
+            dwSize = (lstrlen(L"YES") + 1) * sizeof(WCHAR);
+        else
+            dwSize = sizeof(REG_DWORD);
+
+        if (RegQueryValueExW(hKey,
+                             ClassicOption[iItem].OptionName,
+                             NULL,
+                             &dwType,
+                             (LPBYTE) &chBuffer,
+                             &dwSize) == ERROR_SUCCESS)
+        {
+            if (dwType == REG_SZ)
+            {
+                BOOL stat = ((lstrcmp(chBuffer, L"YES") == 0) || (lstrcmp(chBuffer, L"yes") == 0)) ? TRUE : FALSE;
```

Comment:
`lstrcmpi` would seem better here?
Then there is no need for the duplicate thing.

Reply:
thanks for you review